specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/examples/ManagedEnvironmentsStorages_CreateOrUpdate.json

```diff
@@ -0,0 +1,35 @@
+{
+  "parameters": {
+    "subscriptionId": "8efdecc5-919e-44eb-b179-915dca89ebf9",
+    "resourceGroupName": "examplerg",
+    "envName": "managedEnv",
+    "name": "jlaw-demo1",
+    "api-version": "2022-01-01-preview",
+    "storageEnvelope": {
+      "properties": {
+        "azureFile": {
+          "accountName": "account1",
+          "accountKey": "key",
+          "shareName": "share1",
+          "accessMode": "ReadOnly"
+        }
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/8efdecc5-919e-44eb-b179-915dca89ebf9/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/managedEnv/storages/jlaw-demo1",
```

Comment:
Hey, can you please just explain the scenario a bit here ?

The customer creates the storages passing in the storageaccount key and then you return this resource with the file details. 

- How is this information used by your service ?
- What happens if the storage account key rotates ?

Have you considered using managed identity and accessing the storage using that route ?

Reply:
The storage resource will be reference by the containerapps resource by name. And the service will mount the storage to the container when it is referenced. It is 1:N mapping, so a single file share can be mount to multiple containerapps.
When the account key is rotates, the customer need to update the storage with the latest account information.

For long term, we will support both managed identity and access key based access. But for GA, we will only support access key.